src/folderitemdelegate.cpp

```diff
@@ -45,52 +45,30 @@ FolderItemDelegate::~FolderItemDelegate() {
 
 }
 
+QSize FolderItemDelegate::iconViewTextSize(const QModelIndex& index) const {
+    QStyleOptionViewItem opt;
+    initStyleOption(&opt, index);
+    opt.decorationSize = iconSize_.isValid() ? iconSize_ : QSize(0, 0);
+    opt.decorationAlignment = Qt::AlignHCenter | Qt::AlignTop;
+    opt.displayAlignment = Qt::AlignTop | Qt::AlignHCenter;
+    QRectF textRect(0, 0,
+                    itemSize_.width() - 2 * margins_.width(),
+                    itemSize_.height() - 2 * margins_.height() - opt.decorationSize.height());
+    drawText(nullptr, opt, textRect); // passing nullptr for painter will calculate the bounding rect only
+    return textRect.toRect().size();
+}
+
 QSize FolderItemDelegate::sizeHint(const QStyleOptionViewItem& option, const QModelIndex& index) const {
     QVariant value = index.data(Qt::SizeHintRole);
     if(value.isValid()) {
         // no further processing if the size is specified by the data model
         return qvariant_cast<QSize>(value);
     }
-    // we only handle vertical layout here.
+
     if(option.decorationPosition == QStyleOptionViewItem::Top ||
             option.decorationPosition == QStyleOptionViewItem::Bottom) {
-
-        QStyleOptionViewItem opt = option;
-        // instead of getting decoration size (icon size) from the style option,
-        // we force use of the icon size we specified.
-        QSize decorationSize;
-        if(iconSize_.isValid()) {
-            initStyleOption(&opt, QModelIndex());
-            decorationSize = iconSize_;
-        }
-        else {
-            // FIXME: this operation seems to cause the generation of thumbnail pixmaps just to measure icon sizes.
-            // Since our icon size is a fixed value, we need to avoid this to speed it up.
-            initStyleOption(&opt, index);
-            decorationSize = option.decorationSize;
-        }
-        opt.decorationAlignment = Qt::AlignHCenter | Qt::AlignTop;
-        opt.displayAlignment = Qt::AlignTop | Qt::AlignHCenter;
-
-        // "opt.decorationSize" may be smaller than the requested size because
-        // "QStyledItemDelegate::initStyleOption()" uses "QIcon::actualSize()" to set it
-        // (see Qt -> qstyleditemdelegate.cpp). So, we always get decorationSize from "option".
-        Q_ASSERT(itemSize_ != QSize());
-        QRectF textRect(0, 0, itemSize_.width(), itemSize_.height() - decorationSize.height());
-        drawText(nullptr, opt, textRect); // passing nullptr for painter will calculate the bounding rect only.
-
-        // to draw text shadow, we need one more pixel for it.
-        if(shadowColor_.isValid()) {
```

Comment:
OK, I took a closer look. We have enough space for shadow because we add 2 px at `FolderView::updateGridSize()`.

On the other hand:

(1) Shadow wasn't considered in `FolderItemDelegate::drawText` → `selRect`; so adding 1px for it would have no effect on the space between text and its selection rectangle borders.

(2) Adding 1px for shadow and adjusting `selRect` would *really* have no visible effect other than decreasing the number of desktop items per column, which wouldn't be good at all (we would lose something and gain really nothing).